omegaconf/_utils.py

```diff
@@ -454,6 +444,27 @@ def is_primitive_container(obj: Any) -> bool:
     return is_primitive_list(obj) or is_primitive_dict(obj)
 
 
+def get_union_types(ref_type: Optional[Any]) -> Union[List[Any], Any]:
+    args = getattr(ref_type, "__args__", None)
+    element_types: Union[List[Any], Any]
+    error_msg = ""
+    if args is None:
+        error_msg = "{} is not a valid Union type, it should include more than 1 type."
+    elif Any in args:
+        error_msg = (
+            "{} is not a valid Union type, Any is not a valid union element type."
+        )
+    elif type(None) in args and len(args) == 2:
+        error_msg = "{} is not a valid Union type, we recomend using Optional instead."
+    if error_msg:
+        raise UnsupportedValueType(error_msg.format(ref_type))
+
+    if args is not None and Any not in args:
+        element_types = list(args)
+
+    return element_types
```

Comment:
A few minor tweaks (note the typo fix in `recomend` if you don't copy/paste it exactly):

```suggestion 
    args = getattr(ref_type, "__args__", None)
    error_msg = ""
    if args is None or len(args) < 2:
        error_msg = "{} is not a valid Union type, it should include more than 1 type."
    elif Any in args:
        error_msg = (
            "{} is not a valid Union type, Any is not a valid union element type."
        )
    elif type(None) in args and len(args) == 2:
        error_msg = "{} is not a valid Union type, we recommend using Optional instead."
    if error_msg:
        raise UnsupportedValueType(error_msg.format(ref_type))
```